beamline.rst

```diff
@@ -1,212 +1,379 @@
-Beamline API
-~~~~~~~~~~~~
+General API
+===========
 
-Functionality that incorporates several different instruments or that does
-not necessarily belong to one particular instrument.
+  Functionality (Procedures, Actuators) used generally across the interface.
```

Comment:
This text is a bit vague, all functions and procedures are used generally across the interface.

This part of the interface gathers as I originally wrote:

    "Functionality that incorporates several different instruments or that does not necessarily belong to one particular 
     instrument." 

Each file would in my opinion correspond to one namespace, so that one would do for instance:
    `beamline.run_procedure("QUICK_REALIGN")`

Reply:
There are two question, here, I believe.

- One is whether we should use the same pattern, Enums and NamedTuples for procedures, actuators etc. globally across the specification. I believe we definitely should (in so far as possible), and it was not clear from putting the Actuator class inside the Beamline.py file. If it is wider in scope, it should have a separate file.

- Another is how to package up the interface. That AFAIK is still undecided, and several solutions would work. For now I would tend to propose a single, flat, interface. It is not always obvious which commands should go to which component, especialy since some components are by hardware, and some are by process. 
Is set_actuator_value('Omega', 90.0) a Beamline command, a Diffractometer command, a SampleCentring command? Is set_actuator_value('resolution', 2.0) a Beamline command, or a Detector command? How often (if ever) will we have two procedures with the same name that live in diffferent contexts and do different things?

Apart from that, it does not make too much difference if we call samplechanger.unmount_current_sample(), procedures['sample_changer']['unmount_current_sample'](), or just unmount_current_sample(), as long as we have a clear organisation.

Reply:
I see, 

- My intention is to reuse (when appropriate) the same ActuatorData and ProcedureData throughout the entire API, They were defined here for the moment since it was easier to read in this way (and for now only used here).

- I would say that set_actuator_value('Omega', 90) is either a bemline command or a diffractometer command but not a SampleCentring command (nor a DataCollection command). This since moving omega is not exclusively for sample centering.

I would actually want to place it in diffractometer when I think a bit more about it, but I don't really see any reason against having it in beamline.

- set_actuator_value('resolution', 2.0) is my opinion a property of the beamline (or the beamline configuration) not the detector. 

- So far we will never have concurrent execution of the same type of beamline procedure, they would be limited to thins like, quick realing, centre beam, anneal, scintilator ... 

 - Hm, I would tend to try to suggest some kind of fairly flat namespace that corresponds to the actual beamline layout. The code becomes much clearer to me if we do:

    samplechanger.unmount_current_sample(),

Reply:
Resolution in definitely a beamline command, as it is a function of different equipment.
detector_distance is only a motor but can very well be a beamline and not detector command.
You can set detector distance without being able to set (calculate) the resolution, but frankly does not make any sense.